packages/@aws-cdk/aws-ec2/lib/vpc.ts

```diff
@@ -562,14 +566,15 @@ export interface VpcAttributes {
 
 export interface SubnetAttributes {
   /**
-   * The Availability Zone the subnet is located in
+   * The subnetId for this particular subnet
    */
-  readonly availabilityZone: string;
+  readonly subnetId: string;
 
   /**
-   * The subnetId for this particular subnet
+   * The Availability Zone the subnet is located in
+   * @default - empty string
    */
-  readonly subnetId: string;
+  readonly availabilityZone?: string;
```

Comment:
You are not allowed to make this value optional, because that breaks backwards compatibility.

Regardless, I had rather you change this property into a getter that throws with a descriptive error message if the availability zone is not available. Something like:

```
You cannot reference a Subnet's availability zone if it was not supplied. Add the availabilityZone when importing using Subnet.fromSubnetAttributes()
```